client/reader/user-profile/components/user-profile-header/test/index.tsx

```diff
@@ -0,0 +1,213 @@
+		expect( avatars[ 0 ] ).toHaveAttribute( 'data-author-id', defaultUser.ID.toString() );
+
+		// Test that desktop and mobile versions are properly rendered
+		const desktopAvatar = document.querySelector(
+			'.user-profile-header__avatar-desktop [data-testid="reader-avatar"]'
+		);
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( desktopAvatar ).toBeInTheDocument();
+
+		const mobileAvatar = document.querySelector(
+			'.user-profile-header__avatar-mobile [data-testid="reader-avatar"]'
+		);
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( mobileAvatar ).toBeInTheDocument();
+	} );
+
+	/**
+	 * Test: Display name rendering
+	 *
+	 * Verifies that:
+	 * 1. The user's display name is shown in the header
+	 */
+	test( 'should render the user display name', () => {
+		render( <UserProfileHeader user={ defaultUser } /> );
+
+		// Check if display name is rendered
+		const displayNameEl = screen.getByText( defaultUser.display_name );
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( displayNameEl ).toBeInTheDocument();
+
+		expect(
+			displayNameEl.closest( '[class*="user-profile-header__display-name"]' )
+			// @ts-expect-error -- jest-dom matchers are available globally
+		).toBeInTheDocument();
+	} );
+
+	/**
+	 * Test: Navigation rendering
+	 *
+	 * Verifies that:
+	 * 1. The navigation tabs are rendered with Posts and Lists options
+	 */
+	test( 'should render navigation tabs with Posts and Lists options', () => {
+		render( <UserProfileHeader user={ defaultUser } /> );
+
+		// Check if navigation section is rendered
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( screen.getByTestId( 'section-nav' ) ).toBeInTheDocument();
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( screen.getByTestId( 'nav-tabs' ) ).toBeInTheDocument();
+
+		// Check for navigation items
+		const navItems = screen.getAllByTestId( 'nav-item' );
+		expect( navItems.length ).toBe( 2 ); // Posts and Lists
+
+		// Check nav item content - should have Posts and Lists
+		const navTexts = navItems.map( ( item ) => item.textContent );
+		expect( navTexts ).toContain( 'Posts' );
+		expect( navTexts ).toContain( 'Lists' );
+	} );
+
+	/**
+	 * Test: Bio not rendered when not provided
+	 *
+	 * Verifies that:
+	 * 1. Bio section is not rendered when user has no bio
+	 */
+	test( 'should not render bio section when user has no bio', () => {
+		render( <UserProfileHeader user={ defaultUser } /> );
+
+		// Bio section should not be present
+		const bioSection = document.querySelector( '.user-profile-header__bio' );
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( bioSection ).not.toBeInTheDocument();
+	} );
+
+	/**
+	 * Test: Bio rendering
+	 *
+	 * Verifies that:
+	 * 1. Bio section is rendered when user has a bio
+	 * 2. The bio text is displayed correctly
+	 */
```

Comment:
Should we remove these comments? to me it seems like duplication as the tests itself are  concise and self-explanatory. Moreover, these comments have more chances to become stale so not having these seems better.

Reply:
I find the block-level comments helpful 😅, but I get what you're saying, and I see similar tests don't have them. Removed in 884449e5a89f27c9dd69fcb4f8af0a9ddfe261f1